crates/bevy_remote/src/lib.rs

```diff
@@ -866,3 +961,71 @@ async fn process_single_request(
     let result = result_receiver.recv().await?;
     Ok(BrpResponse::new(request.id, result))
 }
+
+async fn process_brp_websocket(
+    websocket: HyperWebsocket,
+    request_sender: Sender<BrpMessage>,
+    request: BrpRequest,
+) -> AnyhowResult<()> {
+    let mut websocket = websocket.await?;
+
+    let (result_sender, result_receiver) = channel::bounded(1);
+
+    let id = request.id;
+
+    let _ = request_sender
+        .send(BrpMessage {
+            method: request.method,
+            params: request.params,
+            sender: result_sender,
+        })
+        .await;
+
+    while let Ok(result) = result_receiver.recv().await {
+        let response = serde_json::to_string(&BrpResponse::new(id.clone(), result))?;
+        websocket.send(tungstenite::Message::text(response)).await?;
+    }
+
+    Ok(())
+}
+
+fn validate_websocket_request(request: &Request<Incoming>) -> AnyhowResult<BrpRequest> {
+    let body = request
+        .uri()
+        .query()
+        .map(|query| {
+            // Simple query string parsing
+            let mut map = HashMap::new();
```

Comment:
Can we refactor this query string parsing logic to be in its own module so that it can be used in other places as well? It seems like a generically useful piece of logic.